p4-16/spec/P4-16-spec.mdk

```diff
@@ -3408,11 +3408,11 @@ is `0` and the maximum value is `2^W-1`.
 The precedence of saturating addition and subtraction operations is the
 same as for modulo arithmetic addition and subtraction.
 
-All binary operations (except shifts) require both operands to have
+All binary operations (except shifts and concatenation) require both operands to have
 the same exact type and width; supplying operands with different
 widths produces an error at compile time. No implicit casts are
 inserted by the compiler to equalize the widths. There are no binary
-operations that combine signed and unsigned values (except shifts).
+operations that combine signed and unsigned values (except shifts and concatenation).
```

Comment:
Do concatenations support combining signed and unsigned values, without casts?

Reply:
Yes, that is inferred in the original spec:
_8.6.1. Concatenation
Concatenation is applied to two bit-strings (signed or unsigned)... the sign of the result is taken from the left operand._